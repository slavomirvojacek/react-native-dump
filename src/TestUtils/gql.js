const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
