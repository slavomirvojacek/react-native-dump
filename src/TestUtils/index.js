export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
