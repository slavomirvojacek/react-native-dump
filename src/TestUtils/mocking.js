// Sometimes, we might want to test internal state changes without having to render.
// This overrides the setState method to only update the state, without interacting with
// React's internals such as the scheduler.
const mockSetState = (Component) => {
  return class extends Component {
    setState (f) {
      if (typeof f === 'function') {
        this.state = {...this.state, ...f(this.state)}
      } else {
        this.state = {...this.state, ...f}
      }
    }
  }
}

export { mockSetState }
