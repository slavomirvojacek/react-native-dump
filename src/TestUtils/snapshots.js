import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const assertSnapshots = (Component, props, desc, store = null) => {
  let getComponent = p => null

  if (store !== null) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

export { assertSnapshots }
