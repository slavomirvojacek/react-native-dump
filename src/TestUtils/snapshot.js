import React from 'react'
import renderer from 'react-test-renderer'

import { mockSetState } from './mocking'

const assertSnapshot = (C, config) => {
  const {props, state, desc} = config

  let Component

  try {
    const CWithMockState = mockSetState(C)
    Component = new CWithMockState(props)

    if (state) {
      Component.setState(state)
    }
  } catch (e) {
    Component = C(props)
  }

  const RenderedComponent = Component.render ? Component.render() : Component

  test(desc, () => {
    const tree = renderer
      .create(RenderedComponent)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config))
}

export { assertSnapshots }
