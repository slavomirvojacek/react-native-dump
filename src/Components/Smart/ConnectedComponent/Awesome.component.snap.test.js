import React from 'react'
import { Some, None } from '@threestup/monads'

import AwesomeComponent from './Awesome.component'
import { assertSnapshots } from '../../../TestUtils/snapshot'

describe('AwesomeComponent Snapshots', () => {
  const config = [
    {
      props: AwesomeComponent.defaultProps,
      desc: 'renders correctly'
    },
    {
      props: {
        ...AwesomeComponent.defaultProps,
        showLoading: true
      },
      desc: 'renders correctly when loading'
    },
    {
      props: {
        ...AwesomeComponent.defaultProps,
        username: Some('John Doe'),
        showLoading: false
      },
      desc: 'renders correctly when not loading and username present'
    },
    {
      props: {
        ...AwesomeComponent.defaultProps,
        username: None,
        showLoading: false
      },
      desc: 'renders correctly when not loading and username not present'
    },
    {
      props: {
        ...AwesomeComponent.defaultProps,
        error: Some('Err!'),
        showLoading: false
      },
      desc: 'renders correctly when not loading and error present'
    }
  ]

  assertSnapshots(AwesomeComponent, config)
})
