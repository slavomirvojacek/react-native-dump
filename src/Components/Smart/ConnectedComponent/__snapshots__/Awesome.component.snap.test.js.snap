// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwesomeComponent Snapshots renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 20,
      }
    }
  >
    N/A
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when loading 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Loading...
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when not loading and error present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Err!
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when not loading and username not present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 20,
      }
    }
  >
    N/A
  </Text>
</View>
`;

exports[`AwesomeComponent Snapshots renders correctly when not loading and username present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 20,
      }
    }
  >
    John Doe
  </Text>
</View>
`;
